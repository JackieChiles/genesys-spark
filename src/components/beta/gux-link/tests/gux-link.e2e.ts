import { newSparkE2EPage, a11yCheck } from '../../../../../tests/e2eTestUtils';

describe('gux-link-beta', () => {
  describe('#render', () => {
    [
      '<gux-link-beta href="#" link-text="Enabled Link"></gux-link-beta>',
      '<gux-link-beta href="#" link-text="Open in new tab" target="_blank"></gux-link-beta>',
      '<gux-link-beta href="#" link-text="External Link" is-external-link="true" target="_blank"></gux-link-beta>',
      '<gux-link-beta href="#" link-text="Table Cell Link" table-cell-link="true"></gux-link-beta>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-link-beta');

        await a11yCheck(page);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
