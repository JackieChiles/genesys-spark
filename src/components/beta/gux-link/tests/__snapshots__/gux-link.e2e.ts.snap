// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-link-beta #render should render component as expected (1) 1`] = `"<gux-link-beta href=\\"#\\" link-text=\\"Enabled Link\\" hydrated=\\"\\"></gux-link-beta>"`;

exports[`gux-link-beta #render should render component as expected (2) 1`] = `"<gux-link-beta href=\\"#\\" link-text=\\"Open in new tab\\" target=\\"_blank\\" hydrated=\\"\\"></gux-link-beta>"`;

exports[`gux-link-beta #render should render component as expected (3) 1`] = `"<gux-link-beta href=\\"#\\" link-text=\\"External Link\\" is-external-link=\\"true\\" target=\\"_blank\\" hydrated=\\"\\"></gux-link-beta>"`;

exports[`gux-link-beta #render should render component as expected (4) 1`] = `"<gux-link-beta href=\\"#\\" link-text=\\"Table Cell Link\\" table-cell-link=\\"true\\" hydrated=\\"\\"></gux-link-beta>"`;
